react/features/base/settings/reducer.js

```diff
@@ -0,0 +1,163 @@
+
+/**
+ * The default/initial redux state of the feature {@code base/settings}.
+ *
+ * @type Object
+ */
+const DEFAULT_STATE = {};
+
+const STORE_NAME = 'features/base/settings';
+
+/**
+ * Sets up the persistence of the feature {@code base/settings}.
+ *
+ * Note: There are some values inherited from the old Settings.js that needs
+ * session-based storage but not persisted storage, such as cameraDeviceId
+ * and micDeviceId so we utilize the selective storage functionality of the
+ * persistency engine.
+ */
+PersistenceRegistry.register(STORE_NAME, {
+    audioOutputDeviceId: true,
+    avatarID: true,
+    avatarURL: true,
+    cameraDeviceId: false,
+    displayName: true,
+    email: true,
+    localFlipX: true,
+    micDeviceId: false,
+    serverURL: true,
+    startAudioOnly: true,
+    startWithAudioMuted: true,
+    startWithVideoMuted: true
+});
+
+ReducerRegistry.register(STORE_NAME, (state = DEFAULT_STATE, action) => {
+    switch (action.type) {
+    case APP_WILL_MOUNT:
+        return _initSettings(state);
+
+    case SETTINGS_UPDATED:
+        return {
+            ...state,
+            ...action.settings
+        };
+    }
+
+    return state;
+});
+
+/**
+ * Retrieves the legacy profile values regardless of it's being in pre or
+ * post-flattening format.
+ *
+ * @private
+ * @returns {Object}
+ */
+function _getLegacyProfile() {
+    let persistedProfile
+        = window.localStorage.getItem('features/base/profile');
+
+    if (persistedProfile) {
+        try {
+            persistedProfile = JSON.parse(persistedProfile);
+
+            if (persistedProfile && typeof persistedProfile === 'object') {
+                const preFlattenedProfile = persistedProfile.profile;
+
+                return preFlattenedProfile || persistedProfile;
+            }
+        } catch (e) {
+            logger.warn('Error parsing persisted legacy profile', e);
+        }
+    }
+
+    return {};
+}
+
+/**
+ * Inits the settings object based on what information we have available.
+ * Info taken into consideration:
+ *   - Old Settings.js style data
+ *   - Things that we stored in profile earlier but belong here.
+ *
+ * @private
+ * @param {Object} featureState - The current state of the feature.
+ * @returns {Object}
+ */
+function _initSettings(featureState) {
+    let settings = featureState || DEFAULT_STATE;
+
+    // Old Settings.js values
+    const displayName = _.escape(window.localStorage.getItem('displayname'));
+    const email = _.escape(window.localStorage.getItem('email'));
+    let avatarID = _.escape(window.localStorage.getItem('avatarId'));
+
+    if (!avatarID) {
+        // if there is no avatar id, we generate a unique one and use it forever
+        avatarID = randomHexString(32);
+    }
+
+    settings = {
+        avatarID,
+        displayName,
+        email,
+        ...settings
+    };
+
+    if (!browser.isReactNative()) {
+        // Browser only
```

Comment:
also put a FIXME here, these should eventually go away, right?

Reply:
I'm lost now. On my previous note you mentioned that they are not persisted, but they are **here**, since we read them from localStorage, so which one is it? Let's store them in persistency and not in localStorage directly, that why we have persistency, amirite? :-)

Reply:
Oh, sorry, my comment may have been misleading then. These things are persisted, except the cameraDeviceId and micDeviceId, but we agreed recently that those will be persisted too, so I adjusted the code